# KiGuMi
Metagenome hybrid assembly pipeline for a KiGuMi project

Krill has unique enzymatic equipment achieving highly effective hydrolytic breakdown of phyto- and/or zooplankton-derived, complex substrates such as polysaccharides, proteins and lipids. Considering in addition the large biomass of Antarctic krill, Euphausia superba (379 million tons), which exceeds the estimated world fishery catch by a factor of three, there is a high potential for biotechnological exploitation of these hydrolytic enzymes. Surprisingly, the microbiome of the E. superba gut has not been studied in detail and represents a thus far untapped marine bioresource for hydrolytic enzymes. 
The aim of the KiGuMi project is to explore the microbial diversity of the krill gut, establish metaOMICS for the first time in this system, and thereby discover hydrolytic enzymes and microorganisms from the krill gut microbiome that have a high potential for biotechnological exploitation.

Here we present the computational pipeline, used for a hybrid assembly of a krill gut microbiome from the short reads Illumina WGS sequencing data together with the long reads ONT data.
The pipeline utilises a number of standard tools and some custom Perl scripts (see references below).

